### PR TITLE
bpo-34839: Add a 'before 3.6' in the section 'warnings' of doctest

### PR DESCRIPTION
**Pull request by matrixise**
https://bugs.python.org/issue34839

### CODE REVIEW

File: Doc/library/doctest.rst [hunk 1]
```diff
@@ -759,19 +759,19 @@ Warnings
 :mod:`doctest` is serious about requiring exact matches in expected output.  If
 even a single character doesn't match, the test fails.  This will probably
 surprise you a few times, as you learn exactly what Python does and doesn't
-guarantee about output.  For example, when printing a dict, Python doesn't
-guarantee that the key-value pairs will be printed in any particular order, so a
-test like ::
+guarantee about output.  For example, before Python 3.6, when printing a dict,
+Python did not guarantee that the key-value pairs was printed in any
```
Comment (vadmium) [inline Doc/library/doctest.rst]: Plural: pairs _were_ printed

**Review by tim-one — COMMENTED** [review]
As Raymond noted on BPO, it's still the case that nothing about the display order of sets is defined.  So probably better to change the example to use a set of strings instead, and add a note about how the same kinds of things were required for dicts too before 3.6.

**Comment by matrixise** [comment]
Hi @tim-one,

I just updated with your recommendations and those from @rhettinger  .

File: Doc/library/doctest.rst [hunk 2]
```diff
@@ -759,23 +759,27 @@ Warnings
 :mod:`doctest` is serious about requiring exact matches in expected output.  If
 even a single character doesn't match, the test fails.  This will probably
 surprise you a few times, as you learn exactly what Python does and doesn't
-guarantee about output.  For example, when printing a dict, Python doesn't
-guarantee that the key-value pairs will be printed in any particular order, so a
-test like ::
+guarantee about output.  For example, when printing a set, Python doesn't
+guarantee that the key is printed in any particular order, so a test like ::
```
Comment (JulienPalard) [inline Doc/library/doctest.rst]: I would use elements in place of key, for members of a set.

Comment (rhettinger) [reply Doc/library/doctest.rst]: Concur with Julien.  Members of a set are "elements"

**Review by rhettinger — CHANGES_REQUESTED** [review]

File: Doc/library/doctest.rst (hunk 2 above)
Comment (matrixise) [reply Doc/library/doctest.rst]: Hi @JulienPalard and @rhettinger thank you for your catch,

**Comment by matrixise** [comment]
fixed

File: Doc/library/doctest.rst [hunk 3]
```diff
@@ -759,23 +759,27 @@ Warnings
...
-test like ::
+guarantee about output.  For example, when printing a set, Python doesn't
+guarantee that the element is printed in any particular order, so a test like ::
 
    >>> foo()
-   {"Hermione": "hippogryph", "Harry": "broomstick"}
+   {"Hermione", "Harry"}
 
-is vulnerable!  One workaround is to do ::
+was vulnerable!  One workaround is to do ::
```
Comment (andreip) [inline Doc/library/doctest.rst]: maybe keep `is` here? Since sets are still unordered?

Comment (matrixise) [reply Doc/library/doctest.rst]: cool thank you and fixed

**Comment by matrixise** [comment]
Hi @rhettinger 
I have fixed this PR, when you have time, could you check this PR?

Thank you,

Have a nice day,

**Comment by matrixise** [comment]
@rhettinger once you will accept this PR, we could close the issue. ;-)

Thanks

**Review by JulienPalard — APPROVED** [review]

**Pull request merged**